moveit_setup_assistant/src/tools/moveit_config_data.cpp

```diff
@@ -374,6 +374,28 @@ bool MoveItConfigData::outputKinematicsYAML(const std::string& file_path)
   return true;  // file created successfully
 }
 
+// ******************************************************************************************
+// Helper function to get the controller that is controlling the joint
+// ******************************************************************************************
+std::string MoveItConfigData::getJointHardwareInterface(std::string joint_name)
+{
+  for (size_t i = 0; i < ros_controllers_config_.size(); i++)
+  {
+    std::vector<std::string>::iterator joint_it =
+        std::find(ros_controllers_config_[i].joints_.begin(), ros_controllers_config_[i].joints_.end(), joint_name);
+    if (joint_it != ros_controllers_config_[i].joints_.end())
+    {
+      if (ros_controllers_config_[i].type_.substr(0, 6) == "positi")
+        return "hardware_interface/PositionJointInterface";
+      else if (ros_controllers_config_[i].type_.substr(0, 6) == "veloci")
```

Comment:
why only check 6 characters and not 8?

Reply:
yeah you are right, would be cleaner to write the whole word.